### DISCUSSION TITLE
[V3] wire:dirty.remove misbehaves

### DISCUSSION THREAD

**Discussion by livelogic** [livewire/livewire#6668] [7. Bugs (archived)] [answered]
### Livewire version

3.0.1

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Create a Livewire component with one checkbox and two buttons A and B
2. Wire the checkbox via `wire:model="prop"`
2. On button A, set attribute `wire:dirty`
3. On button B, set attribute `wire:dirty.remove`
4. Test the component:  button A is initially hidden, button B is visible
5. Tick the checkbox: button A appears, button B hides
6. Tick the checkbox again: button A hides, button B remains hidden

### Additional information

Example template:
```
<input type="checkbox" wire:model="prop" />
<button wire:click="clicked" wire:dirty>A</button>
<button wire:click="clicked" wire:dirty.remove>B</button>
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@livelogic thanks for reporting! Any chance you could submit a failing test PR? Thanks!

**Reply by livelogic** [reply]
@joshhanley I'm sorry i don't have tests for this issue, I only noticed it while testing my apps after upgrading to v3.

**Reply by joshhanley** [reply]
@livelogic Yep, I'm asking if you could write a failing test for Livewire and submit it as a PR. We're swapped with issues, so a failing test PR is the easiest way for us to fix the issue 🙂

**Reply by livelogic** [reply]
@joshhanley I'm afraid i can't help with that. I believe the task requires Dusk tests, hopefully someone better equipped will contribute :)

**Comment by PhiloNL** [comment, marked as answer]
Thanks for reporting. Added failing test https://github.com/livewire/livewire/pull/6759 :)
